bundles/org.openhab.binding.dominoswiss/src/main/java/org/openhab/binding/dominoswiss/internal/EGateHandler.java

```diff
@@ -0,0 +1,321 @@
+        sendCommand("Instruction=1;ID=" + id + ";Command=5;Priority=1;CheckNr=5408219;" + CR);
+    }
+
+    // @SuppressWarnings("null")
+    public void registerBlind(String id, ThingUID uid) {
+        logger.debug("Registring Blind id {} with thingUID {}", id, uid);
+        registeredBlinds.put(id, uid);
+    }
+
+    /**
+     * Send a command to the eGate Server.
+     */
+
+    private void sendCommand(String command) {
+        sendCommand(command, SOCKET_TIMEOUT_SEC);
+    }
+
+    @SuppressWarnings("null")
+    private synchronized void sendCommand(String command, int timeout) {
+        logger.debug("EGate got command: {}", command);
+
+        if (!isConnected()) {
+            logger.debug("no connection to Dominoswiss eGate server when trying to send command, returning...");
+            return;
+        }
+
+        // Send plain string to eGate Server,
+        try {
+            egateSocket.setSoTimeout(SOCKET_TIMEOUT_SEC);
+            writer.write(command);
+            writer.flush();
+        } catch (IOException e) {
+            logger.error("Error while sending command {} to Dominoswiss eGate Server {} ", command, e.toString());
+        }
+    }
+
+    private void pollingConfig() {
+        if (!isConnected()) {
+            synchronized (lock) {
+                try {
+                    egateSocket = new Socket();
+                    logger.debug("before connect");
+                    egateSocket.connect(new InetSocketAddress(host, port));
+                    logger.debug("after connect");
+                    egateSocket.setSoTimeout(SOCKET_TIMEOUT_SEC);
+                    logger.debug("before writer");
+                    writer = new BufferedWriter(new OutputStreamWriter(egateSocket.getOutputStream()));
+                    writer.write("SilenceModeSet;Value=0;" + CR);
+                    writer.flush();
+
+                } catch (IOException e) {
+                    logger.debug("IOException in pollingConfig: {} host {} port {}", e.toString(), host, port);
+                    try {
+                        egateSocket.close();
+                        egateSocket = null;
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.toString());
+                    } catch (IOException e1) {
+                        logger.debug("EGate Socket not closed {}", e1.toString());
+                    }
+                    egateSocket = null;
+                }
+                if (egateSocket != null) {
+                    updateStatus(ThingStatus.ONLINE);
+                    startAutomaticRefresh();
+                    logger.debug("EGate Handler started automatic refresh, status: {} ",
+                            getThing().getStatus().toString());
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                    logger.debug("Something went wrong: Host {} Port {}", host, port);
```

Comment:
The log message could be removed, as the status update is already logged by the framework. You can integrate the message in `updateStatus()` to display it in the UI.

The state change originated by updateStatus() is logged to events.log. Including the status detail message.

Please check all.